onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -174,6 +174,13 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
 
     if (converted_to_constant) {
+      // Remove single-output node chain for inputs of the node
+      for (int j = 0; j < node->InputArgCount().front(); ++j) {
```

Comment:
for (auto iter = node->InputNodesBegin(); iter != node->InputNodesEnd(); ++iter)